client/state/analytics/actions.js

```diff
@@ -0,0 +1,47 @@
+import fromPairs from 'lodash/fromPairs';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants.js';
+
+export const withAnalytics = ( analytics, action ) =>
+	Object.assign( {}, action, { meta: Object.assign( {}, action.meta, analytics.meta ) } );
+
+export const bumpStat = ( group, name ) => ( {
+	type: BUMP_STAT,
+	meta: {
```

Comment:
Well, this can get complicated, but we should either rename `meta` to `analytics-meta` or something similar OR make meta an array.
Because we can later decide on adding further "special" abilities to actions for different purposes...